integrations/wiz/client.py

```diff
@@ -0,0 +1,246 @@
+]
+
+ISSUES_GQL = """
+query IssuesTable(
+  $filterBy: IssueFilters
+  $first: Int
+  $after: String
+  $orderBy: IssueOrder
+) {
+  issues: issuesV2(
+    filterBy: $filterBy
+    first: $first
+    after: $after
+    orderBy: $orderBy
+  ) {
+    nodes {
+      id
+      sourceRule {
+        __typename
+        ... on Control {
+          id
+          name
+          controlDescription: description
+          resolutionRecommendation
+          securitySubCategories {
+            title
+            category {
+              name
+              framework {
+                name
+              }
+            }
+          }
+        }
+        ... on CloudEventRule {
+          id
+          name
+          cloudEventRuleDescription: description
+          sourceType
+          type
+        }
+        ... on CloudConfigurationRule {
+          id
+          name
+          cloudConfigurationRuleDescription: description
+          remediationInstructions
+          serviceType
+        }
+      }
+      createdAt
+      updatedAt
+      dueAt
+      type
+      resolvedAt
+      statusChangedAt
+      projects {
+        id
+        name
+        slug
+        businessUnit
+        riskProfile {
+          businessImpact
+        }
+      }
+      status
+      severity
+      entitySnapshot {
+        id
+        type
+        nativeType
+        name
+        status
+        cloudPlatform
+        cloudProviderURL
+        providerId
+        region
+        resourceGroupExternalId
+        subscriptionExternalId
+        subscriptionName
+        subscriptionTags
+        tags
+        createdAt
+        externalId
+      }
+      serviceTickets {
+        externalId
+        name
+        url
+      }
+      notes {
+        createdAt
+        updatedAt
+        text
+        user {
+          name
+          email
+        }
+        serviceAccount {
+          name
+        }
+      }
+    }
+    pageInfo {
+      hasNextPage
+      endCursor
+    }
+  }
+}
+"""
+
+
+class TokenResponse(BaseModel):
+    access_token: str = Field(alias="access_token")
+    expires_in: int = Field(alias="expires_in")
+    token_type: str = Field(alias="token_type")
+    _retrieved_time: int = PrivateAttr(get_time())
+
+    @property
+    def expired(self) -> bool:
+        return self._retrieved_time + self.expires_in < get_time()
+
+    @property
+    def full_token(self) -> str:
+        return f"{self.token_type} {self.access_token}"
+
+
+class CacheKeys(StrEnum):
+    ISSUES = "wiz_issues"
+
+
+class WizClient:
+    def __init__(
+        self, api_url: str, client_id: str, client_secret: str, token_url: str
+    ):
+        self.api_url = api_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_url = token_url
+        self.last_token_object: TokenResponse | None = None
+
+        self.http_client = http_async_client
+
+    @property
+    async def api_auth_headers(self) -> dict[str, Any]:
```

Comment:
please sort the functions in a reasonable way to read

Reply:
I have now